module/VuFind/src/VuFind/Http/CachingDownloader.php

```diff
@@ -87,21 +73,26 @@ class CachingDownloader implements \VuFindHttp\HttpServiceAwareInterface
      *
      * @param CacheManager  $cacheManager  VuFind Cache Manager
      * @param ConfigManager $configManager VuFind Config Manager
+     * @param bool          $cacheEnabled  Main toggle for enabling caching
      */
-    public function __construct(CacheManager $cacheManager, ConfigManager $configManager)
-    {
-        $this->cacheManager = $cacheManager;
-        $this->configManager = $configManager;
+    public function __construct(
+        protected CacheManager $cacheManager,
+        protected ConfigManager $configManager,
+        protected bool $cacheEnabled
+    ) {
         $this->setUpCache('default');
     }
 
     /**
      * Get cache and initialize it, if necessary.
      *
-     * @return StorageInterface
+     * @return ?StorageInterface
```

Comment:
Maybe add "Return null if cache is disabled." to the description above.

Reply:
This okay?
`     * @return ?StorageInterface Cache storage interface or null if disabled`